### PR TITLE
Add sphinx for generating documentation

### PR DESCRIPTION
**Pull request by hakbailey**
#### What does this PR do?
Adds Sphinx , which is the standard Python tool for generating detailed documentation from docstrings. We will use it to generate and update the official docs as we develop and maintain the library.

* Adds Sphinx, theme, and extras to dev dependencies
* Creates a docs folder with initial Sphinx config and files
* Updates README format for easy inclusion in Sphinx-generated docs
* Creates a make command to build docs with Sphinx
* Updates PR template to include Sphinx in developer checklist

#### How can a reviewer manually see the effects of these changes?
Install new dependencies (`poetry install`) and then run `make sphinx` locally to build the docs. Open `docs/_build/html/index.html` in a browser to view the docs.

NOTE: all changes are in the second commit in this PR. This branch will be rebased when PR #2 is merged. DONE

#### Includes new or updated dependencies?
YES

#### Side effects of this change:
None, but a new ticket has been created to handle publishing the docs (ETD-420, linked below).

#### Relevant ticket(s):
* https://mitlibraries.atlassian.net/browse/ETD-392
* https://mitlibraries.atlassian.net/browse/ETD-420

#### Developer
- [ ] Docstrings and README are updated to reflect all changes as needed
- [ ] Sphinx source docs are updated to reflect all changes as needed
- [ ] All related Jira tickets are linked in commit message(s)

#### Code Reviewer
- [x] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [x] There are appropriate tests covering any new functionality
- [x] The documentation has been updated to reflect all changes or is not needed
- [x] The changes have been verified
- [x] New dependencies are appropriate or there were no changes

### CODE REVIEW

**Comment by hakbailey** [comment]
Force push to rebase onto updated main branch.

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import os
+import sys
+
+sys.path.insert(0, os.path.abspath(".."))
+
+
+# -- Project information -----------------------------------------------------
+
+project = "dspace-python-client"
+copyright = "2021, Helen Bailey, Eric Hanson"
```
Comment (JPrevost) [inline docs/conf.py]: I usually put MIT Libraries as the copyright owner in the license files I create. No idea if that is correct but I'm pretty sure we don't retain copyright on works for hire without an explicit contract stating we do (which we don't have).

File: README.rst [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+You only need to install Poetry once, so if you've already installed it for another project you can skip this step. Poetry will automatically pick up the current Python version and use it to create virtualenvs accordingly.
+
+To install poetry:
+
+* Mac/unix::
+
+    curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python -
+
+* Windows::
```
Comment (JPrevost) [inline README.rst:39]: I'd probably remove the windows instructions unless you've actually tested them or maybe change them to a more generic "google and good luck" kinda thing as that syntax looks really weird (I'm sure it's accurate but also... 🤷🏻 )

This looks pretty boilerplate-y so also feel free to keep it if ya want of course :)

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+^^^^^^^^^^^^
+Installation
+^^^^^^^^^^^^
+
+Install dspace-python-client for local development by running::
+
+  git clone https://github.com/MITLibraries/dspace-python-client.git
+  cd dspace-python-client
+  make install
```
Comment (JPrevost) [inline README.rst:55]: We chatted on slack but at some point we may want to document how to use the correct python version (or at least gently remind that if you don't have python versioning working locally that any issues are out of scope of this project to solve).

File: README.rst [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+~~~~~~
+Poetry
+~~~~~~
+This project uses Poetry for dependency management and packaging.
+
+You only need to install Poetry once, so if you've already installed it for another project you can skip this step. Poetry will automatically pick up the current Python version and use it to create virtualenvs accordingly.
+
+To install poetry:
+
+* Mac/unix::
```
Comment (JPrevost) [inline README.rst:35]: FYI I used `brew install poetry`. I'm not sure if the default curl install handles updates but I know brew will so I felt better about that path.

**Review by JPrevost — COMMENTED** [review]
I suspect we should update the copyright but the rest (and even that honestly) isn't blocking. If you prefer not to make any changes let me know and I'll approve as-is.

File: docs/conf.py (hunk 1 above)
Comment (hakbailey) [reply docs/conf.py]: Oh good catch, thank you! Will update.

File: README.rst (hunk 2 above)
Comment (hakbailey) [reply README.rst:39]: I haven't tested them but they're copied directly from the Poetry docs...

File: README.rst (hunk 3 above)
Comment (hakbailey) [reply README.rst:55]: Yeah I'm not opposed to this although 1) I'm hoping the addition of the .python-version file in a future PR will help, and 2) I'd rather do it in a future PR if we run into more issues.

File: README.rst (hunk 4 above)
Comment (hakbailey) [reply README.rst:35]: The poetry docs explicitly recommend against that. I wonder if that's part of what caused the python version issues? 

```
Using alternative installation methods will make Poetry always use the Python version for which it has been installed to create virtualenvs.

So, you will need to install Poetry for each Python version you want to use and switch between them.```

Comment (JPrevost) [reply README.rst:35]: Yeah that's definitely not how it seems to be working for me. I didn't need to reinstall poetry, I just needed to switch to the version of python I wanted and then it worked. I think with the dot file for the python version I wouldn't have run into this. Documenting the official path is fine... still curious if it handles updates but either way 👍🏻

**Review by JPrevost — APPROVED** [review]
Feel free to add in the two missing generated docs files to this branch before merge or just include in another one. Either way :shipit:
